cub/cub/detail/binary_search_helpers.cuh

```diff
@@ -0,0 +1,65 @@
+// SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+
+#include <cub/config.cuh>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/__algorithm/lower_bound.h>
+#include <cuda/std/__algorithm/upper_bound.h>
+#include <cuda/std/tuple>
+
+CUB_NAMESPACE_BEGIN
+
+namespace detail::find
+{
+template <typename RangeIteratorT, typename CompareOpT, typename Mode>
+struct comp_wrapper_t
+{
+  RangeIteratorT first;
+  RangeIteratorT last;
+  CompareOpT op;
+
+  template <typename Value, typename Output>
+  _CCCL_DEVICE _CCCL_FORCEINLINE void operator()(cuda::std::tuple<Value, Output> args) const
```

Comment:
Important: Please fully qualify all references to entities in the `cuda` namespace:
```suggestion
  _CCCL_DEVICE _CCCL_FORCEINLINE void operator()(::cuda::std::tuple<Value, Output> args) const
```
Applies to several other places in this PR.